e2e/test/Helpers/TestDevice.cs

```diff
@@ -82,14 +84,18 @@ private static async Task<TestDevice> CreateDeviceAsync(TestDeviceType type, str
                 auth = new DeviceAuthenticationWithX509Certificate(deviceName, Configuration.IoTHub.GetCertificateWithPrivateKey());
             }
 
-            Device device = await rm.AddDeviceAsync(requestDevice).ConfigureAwait(false);
-
-            _logger.Trace($"{nameof(GetTestDeviceAsync)}: Pausing for {DelayAfterDeviceCreationSeconds}s after device was created.");
-            await Task.Delay(DelayAfterDeviceCreationSeconds * 1000).ConfigureAwait(false);
+            Device device = null;
+            await RetryOnThrottlingAsync(
+                async () =>
+                {
+                    device = await rm.AddDeviceAsync(requestDevice).ConfigureAwait(false);
+                });
```

Comment:
.ConfigureAwait(false)

Reply:
I am not sure what you mean here

Reply:
await RetryOnThrottlingAsync(....)..ConfigureAwait(false)
I meant .ConfigureAwait(false) is missing

Reply:
Ah, got it, I thought you were talking about the r..AddDevicesAsync API!
Will add.